src/Orleans/CodeGeneration/GrainInterfaceData.cs

```diff
@@ -145,22 +145,118 @@ public static bool IsGrainReference(Type t)
+
+            return methodInfos.ToArray();
+        }
+
+        /// <summary>
+        /// Recurses through entire interface graph accumulating methods.
+        /// </summary>
+        /// <param name="interfaceType"></param>
+        /// <param name="methodInfos"></param>
+        /// <param name="bIncludeAllParentsMethods"></param>
+        private static void GetInterfaceMethods(System.Type interfaceType, ref List<MethodInfo> methodInfos, bool bIncludeAllParentsMethods)
         {
-            var methodInfos = new List<MethodInfo>();
-            GetMethodsImpl(grainType, grainType, methodInfos);
-            var flags = BindingFlags.Public | BindingFlags.InvokeMethod | BindingFlags.Instance;
-            if (!bAllMethods)
-                flags |= BindingFlags.DeclaredOnly;
+            if (!interfaceType.IsInterface)
+            {
+                throw new ArgumentException("Illegal argument: interfaceType. Must be an interface.");
+            }
+
+            //Default flags (BindingFlags.Public) for GetMethod works fine here given that interfaceType is an interface and not a class (Hence .Invoke and .Instance flags are not required).
+            MethodInfo[] infos = interfaceType.GetMethods();
 
-            MethodInfo[] infos = grainType.GetMethods(flags);
             IEqualityComparer<MethodInfo> methodComparer = new MethodInfoComparer();
             foreach (var methodInfo in infos)
                 if (!methodInfos.Contains(methodInfo, methodComparer))
                     methodInfos.Add(methodInfo);
+            
+            if (bIncludeAllParentsMethods)
+            {
+                var parentInterfaces = interfaceType.GetInterfaces();
+                foreach (var parent in parentInterfaces)
+                {
+                    //no need for recursion as .GetInterfaces() actually returns all parent interfaces at all levels.
+                    //GetInterfaceMethods(parent, ref methodInfos, bIncludeAllParentsMethods);
+                    infos = parent.GetMethods();
+                    foreach (var methodInfo in infos)
+                        if (!methodInfos.Contains(methodInfo, methodComparer))
+                            methodInfos.Add(methodInfo);        
+                }
+            }
+        }
 
-            return methodInfos.ToArray();
+        //See explanation for GetGrainInterfaceMethods above.
+        /// <summary>
+        /// Returns all the properties declared by the grain type and it's parents in the inheritance tree.
+        /// </summary>
+        /// <param name="grainInterfaceType">"grainInterfaceType" must be an interface (not a class) and must be inherited from IGrain at some level.</param>
+        /// <param name="bIncludeAllParentsMethods"></param>
+        /// <returns></returns>
+        public static PropertyInfo[] GetGrainInterfaceProperties(Type grainInterfaceType, bool bIncludeAllParentsMethods = false)
```

Comment:
We don't support properties on grain interfaces (for about a year, we used to). So this code is unnecessary.